sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -30,13 +30,16 @@ export class DefaultAzureCredential extends ChainedTokenCredential {
     credentials.push(new EnvironmentCredential(tokenCredentialOptions));
     credentials.push(new ManagedIdentityCredential(tokenCredentialOptions));
     if (process.env.AZURE_CLIENT_ID) {
-      credentials.push(new ManagedIdentityCredential(process.env.AZURE_CLIENT_ID, tokenCredentialOptions));
+      credentials.push(
+        new ManagedIdentityCredential(process.env.AZURE_CLIENT_ID, tokenCredentialOptions)
+      );
     }
     credentials.push(new AzureCliCredential());
     credentials.push(new VSCodeCredential(tokenCredentialOptions));
 
-    super(
-      ...credentials
-    );
+    super(...credentials);
+    this.AuthenticationFailedMessage = "DefaultAzureCredential authentication failed";
```

Comment:
if all that is different in the message is the class name, why not construct it dynamically?